exasol/nb_connector/text_ai_extension_wrapper.py

```diff
@@ -124,21 +153,37 @@ def deploy_licence(
             Optional. Content of a licence given as a string.
 
     """
-    raise NotImplementedError(
-        "Currently this is not implemented, "
-        "will be changed once the licensing process is finalized."
-    )
+
+    license_content = license_content or txai_licenses.COMMUNITY_LICENSE
+    # if license_content:
+    #     pass
+    # elif license_file:
+    #     license_content = license_file.read_text()
+    # else:
+    #     license_content = txai_licenses.COMMUNITY_LICENSE
+
+    pyexasol_connection = open_pyexasol_connection(conf)
+    txai_licenses.create_connection(pyexasol_connection, license_content)
+
+
+@dataclass
+class InstallOptions:
+    slc: bool = True
+    udf_scripts: bool = False
+    models: bool = False
+    bfs_credentials: bool = True
 
 
 def initialize_text_ai_extension(
     conf: Secrets,
     container_file: Optional[Path] = None,
     version: Optional[str] = None,
     language_alias: str = LANGUAGE_ALIAS,
-    run_deploy_container: bool = True,
-    run_deploy_scripts: bool = False,
-    run_upload_models: bool = False,
-    run_encapsulate_bfs_credentials: bool = True,
+    run_deploy_container: bool = True,  # proposal: rename to install_slc
```

Comment:
Optional refactoring.

As this will break the interface in the Jupyter notebooks, I wanted to ask whether this is acceptable. 
Also, the interface will then differ from the wrappers of other extensions.

Reply:
we need to adapt it anyway, however, I would keep the parameter and don't make a object out of them